src/swarm/neo/protocol/ProtocolError.d

```diff
@@ -96,17 +96,20 @@ class ProtocolError: Exception
     private void setFmt_ ( T... ) ( istring file, long line, cstring fmt, T args )
     {
         this.reused_msg.reset();
-        sformat(
-            cast(FormatterSink) (cstring chunk) {
-                this.reused_msg ~= chunk;
-                return chunk.length;
-            },
-            fmt,
-            args
-        );
+        static if (is(FormatterSink))
```

Comment:
Does this depend on a certain ocean version?

Reply:
Yes, `FormatterSink` was introduced with the deprecation of the `size_t` overload.

Reply:
I see. I was wondering if it might be clearer to statically check for the library version.

Reply:
I can't do that, because `LibFeature` wasn't properly updated at the time (that was found when I tried to do a similar change in another library).

Reply:
Ah ok, fair enough.

That makes me wonder: do we have a policy for removing lib-version based `static if`s at some point? We probably should. It'd be nice to make sure (via a comment?) that this conditional gets removed during such a purge.

Reply:
What do you think about this?

Reply:
> That makes me wonder: do we have a policy for removing lib-version based static ifs at some point? We probably should. It'd be nice to make sure (via a comment?) that this conditional gets removed during such a purge.

We don't at the moment, because no one have been using them before, so the issue hasn't come up. I added a comment (and make sure to include the word "deprecated" so grep will show it).

Reply:
:+1: I missed the extra comment before.

Reply:
No you didn't, I missed your comment :P